designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Button.kt

```diff
@@ -46,18 +64,118 @@ fun SquareIconButton(
+        IconStyle.OUTLINED -> CustomOutlinedIconButton(enabled, icon, onClick)
+        else -> StandardIconButton(enabled, icon, onClick)
+    }
+}
+
+@Composable
+fun StandardIconButton(
+    enabled: Boolean = true,
+    icon: @Composable (() -> Unit),
+    onClick: () -> Unit
+) {
+    CompositionLocalProvider(LocalRippleTheme provides CustomDHISRippleTheme) {
+        FilledIconButton(
+            onClick = onClick,
+            modifier = Modifier
+                .size(Spacing.Spacing48)
+                .padding(Spacing.Spacing4),
+            enabled = enabled,
+            colors = IconButtonDefaults.iconButtonColors(Color.Transparent, TextColor.OnSurfaceVariant, Color.Transparent, TextColor.OnDisabledSurface)
+        ) {
+            icon()
+        }
+    }
+}
+
+@Composable
+fun CustomFilledIconButton(
+    enabled: Boolean = true,
+    icon: @Composable (() -> Unit),
+    onClick: () -> Unit
+) {
+    FilledIconButton(
+        onClick = onClick,
+        modifier = Modifier
+            .size(Spacing.Spacing48)
+            .padding(Spacing.Spacing4),
         enabled = enabled,
-        onClick = onClick
+        colors = IconButtonDefaults.iconButtonColors(SurfaceColor.Primary, TextColor.OnPrimary, SurfaceColor.DisabledSurface, TextColor.OnDisabledSurface)
     ) {
         icon()
     }
 }
+
+@Composable
+fun CustomFilledTonalIconButton(
+    enabled: Boolean = true,
+    icon: @Composable (() -> Unit),
+    onClick: () -> Unit
+) {
+    CompositionLocalProvider(LocalRippleTheme provides CustomDHISRippleTheme) {
+        FilledTonalIconButton(
+            onClick = onClick,
+            modifier = Modifier
+                .size(Spacing.Spacing48)
+                .padding(Spacing.Spacing4),
+            enabled = enabled,
+            shape = CircleShape,
+            colors = IconButtonDefaults.filledTonalIconButtonColors(SurfaceColor.PrimaryContainer, TextColor.OnPrimaryContainer, SurfaceColor.DisabledSurface, TextColor.OnDisabledSurface)
+
+        ) {
+            icon()
+        }
+    }
+}
+
+@Composable
+fun CustomOutlinedIconButton(
+    enabled: Boolean = true,
+    icon: @Composable (() -> Unit),
+    onClick: () -> Unit
+) {
+    CompositionLocalProvider(LocalRippleTheme provides CustomDHISRippleTheme) {
+        OutlinedIconButton(
+            onClick = onClick,
+            modifier = Modifier
+                .size(Spacing.Spacing48)
+                .padding(Spacing.Spacing4),
+            enabled = enabled,
+            shape = CircleShape,
+            border = BorderStroke(Spacing.Spacing1, if (enabled) Outline.Dark else SurfaceColor.DisabledSurface),
+            colors = IconButtonDefaults.outlinedIconButtonColors(Color.Transparent, TextColor.OnPrimaryContainer)
+        ) {
+            icon()
+        }
+    }
+}
+
+enum class IconStyle {
+    STANDARD, FILLED, TONAL, OUTLINED
+}
+
+object CustomDHISRippleTheme : RippleTheme {
```

Comment:
is this going to be used in another places? What do you think about extracting it to a custom class inside theme package called Ripple.kt or something that fits to you?